sklearn/neighbors/tests/test_kde.py

```diff
@@ -141,3 +141,44 @@ def test_kde_pipeline_gridsearch():
     search = GridSearchCV(pipe1, param_grid=params, cv=5)
     search.fit(X)
     assert_equal(search.best_params_['kerneldensity__bandwidth'], .1)
+
+
+def test_kde_sample_weights():
+    N = 10000
+    T = 100
+    W_neutral = 3 * np.ones(N)
+    for d in [1, 2, 10]:
+        rng = np.random.RandomState(0)
+        X = rng.rand(N, d)
+        W = 1 + (10 * X.sum(axis=1)).astype(np.int8)
```

Comment:
`W = rng.randint(10, size=N)` is probably clearer.

Reply:
Indeed, however the goal is not to have weights uniformly distributed, since this would be (asymptotically) equivalent to having uniform weights.
So I picked weights to be entirely determined the (L1) norm of the vector to have a simple pattern where weights are positive integers.